### PR TITLE
Correct docstring.

### LINKED ISSUE #8837: Inconsistent PreTrainedTokenizerBase.pad  argument default value & docstring
The docstring states the argument `padding` has a default of `False` but its default is `True`

docstring: 
https://github.com/huggingface/transformers/blob/d5b3e56de5376aa85ef46e7f0325139d9e299a41/src/transformers/tokenization_utils_base.py#L2469-L2470
arg:
https://github.com/huggingface/transformers/blob/d5b3e56de5376aa85ef46e7f0325139d9e299a41/src/transformers/tokenization_utils_base.py#L2431-L2472

This causes issues when using `DataCollatorForLanguageModeling` with an already padded dataset as it resets the attention mask.

### PR DESCRIPTION
**Pull request by Fraser-Greenlee**
Related issue: https://github.com/huggingface/transformers/issues/8837

# What does this PR do?

Updating the PreTrainedTokenizerBase.pad argument default value docstring to show the correct default value.

**Current**
docstring: 
https://github.com/huggingface/transformers/blob/d5b3e56de5376aa85ef46e7f0325139d9e299a41/src/transformers/tokenization_utils_base.py#L2469-L2470
arg:
https://github.com/huggingface/transformers/blob/d5b3e56de5376aa85ef46e7f0325139d9e299a41/src/transformers/tokenization_utils_base.py#L2431-L2472

Fixes #8837 (issue)

I'm also curious why this method has default `padding=True`? Other methods (prepare_for_model, encode, __call__, encode_plus, batch_encode_plus) have `padding=False`.

Its default means the DataCollatorForLanguageModeling pads input examples which means it can't be simply switched with the default collator in the [example script](https://github.com/huggingface/transformers/blob/master/templates/adding_a_new_example_script/%7B%7Bcookiecutter.directory_name%7D%7D/run_%7B%7Bcookiecutter.example_shortcut%7D%7D.py#L287-L306) without breaking the attention mask.
https://github.com/huggingface/transformers/blob/610cb106a216cfb99d840648b576f9502189e4d1/src/transformers/data/data_collator.py#L253

@mfuntowicz
@LysandreJik 
@sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for fixing!
